src/charts/BarChart.ts

```diff
@@ -0,0 +1,98 @@
+import { Chart } from 'chart.js'
+import { createCanvas } from 'canvas'
+import { customCanvasBackgroundColor } from './plugins'
+
+const RED = 'rgba(255, 99, 132)'
+const GREEN = 'rgba(75, 192, 192)'
+
+export interface BarChartEntry {
+  yValue: number
+  xValue: number
+  barLabel: string
+  label?: string
+}
+
+interface BarOptions {
+  magicNumber: number
+}
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export default class BarChart {
+  static {
+    Chart.defaults.font.weight = 'bold'
+    Chart.defaults.font.size = 24
+  }
+
+  public static getImage (
+    entries: BarChartEntry[],
+    width = 2000,
+    height = 1000,
+    opts: BarOptions = { magicNumber: 2 }
+  ): string {
+    const canvas = createCanvas(width, height)
+    const ctx = canvas.getContext('2d')
+
+    const chart = new Chart(ctx, {
+      type: 'bar',
+      data: {
+        labels: entries.map((entry) => entry.xValue),
+        datasets: [{
+          data: entries.map((entry) => ({ y: entry.yValue, x: entry.xValue, label: entry.barLabel })),
+          backgroundColor: entries.map((row) => row.xValue <= opts.magicNumber ? RED : GREEN),
+          borderColor: entries.map((row) => row.xValue <= opts.magicNumber ? RED : GREEN),
+          borderWidth: 1
+        }]
+      },
+      options: {
+        elements: {
+          bar: {
+            borderRadius: 10
+          }
+        },
+        plugins: {
+          legend: {
+            display: true,
+            labels: {
+              generateLabels: (_: Chart) => [
+                { text: 'low replica count', fillStyle: RED, strokeStyle: '#fff' },
```

Comment:
Let's move all those customizable text to options

Reply:
Also I started to use Number of Providers instead of Replicas to reduce confusion. We also need to change the corresponding labels on the chart